source/auth/auth.rst

```diff
@@ -87,6 +87,14 @@ mechanism (string)
 mechanism_properties
 	* Includes additional properties for the given mechanism.
 
+Ambiguity between authentication source and URI database
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The presence of a database name in the URI connection string MUST NOT be
+interpreted as a user configuring authentication credentials.  The URI database
```

Comment:
Does this warrant a new spec test in https://github.com/mongodb/specifications/blob/master/source/auth/tests/connection-string.yml? 

I'll note that [`auth/tests`](https://github.com/mongodb/specifications/tree/master/source/auth/tests) doesn't have a README but I assume it uses the format in [`connection-string/tests/README.rst`](https://github.com/mongodb/specifications/blob/master/source/connection-string/tests/README.rst). The connection string's test README doesn't explicitly say that `auth` may be null (although it clearly is in many tests) and we also don't have any tests for soliciting a default database name without UserInfo specified.

I think a test like the following could accomplish this:

```yaml
    -
        description: "default database without credentials does not infer auth"
        uri: "mongodb://localhost/test"
        hosts: ~
        valid: true
        warning: false
        auth: ~
        options: ~
```

Not sure if this belongs in `auth`, `connection-string`, or both.

Reply:
Done in 3fffcae

Reply:
The connection string test README says that null value for a key like `auth` means no assertion is necessary, so that test wouldn't verify anything.  :-(